packages/amplify-category-function/src/provider-utils/awscloudformation/utils/functionPluginLoader.ts

```diff
@@ -162,6 +149,18 @@ async function getSelectionsFromContributors<T>(
     .reduce((acc, it) => acc.concat(it), [])
     .sort((a, b) => a.name.localeCompare(b.name)); // sort by display name so that selections order is deterministic
 
+  let prevruntime;
+  if (selectionOptions.runtimeState !== undefined) {
+    prevruntime = selections
+      .filter(runtime => {
+        for (let val of selectionOptions.runtimeState) {
+          if (val === runtime.name) return runtime;
+        }
+      })
+    prevruntime.forEach(selection =>{
+      _.assign(selection,{checked : true})
+    });
+  }
```

Comment:
this should be it's own function and be called from within the else block below. It looks like you're trying to set {checked: true} on the selections that appear in runtimeState in which case I think all of this simplifies to
```js
selections
  .filter(selection => selectionOptions.runtimeState.includes(selection.name))
  .forEach(selection => _.assign(selection, {checked: true})
```

Reply:
Thanks for this Edward. Done!!